examples/ct/handlers.go

```diff
@@ -100,454 +122,362 @@ type LogContext struct {
+	return addChainInternal(ctx, c, w, r, true)
 }
 
-func getSTH(c LogContext, w http.ResponseWriter, r *http.Request) (int, error) {
-	if !enforceMethod(w, r, http.MethodGet) {
-		return http.StatusMethodNotAllowed, fmt.Errorf("method not allowed: %s", r.Method)
+func getSTH(ctx context.Context, c LogContext, w http.ResponseWriter, r *http.Request) (int, error) {
+	// Forward on to the Log server.
+	req := trillian.GetLatestSignedLogRootRequest{LogId: c.logID}
+	if glog.V(2) {
+		glog.Infof("%s: GetSTH => grpc.GetLatestSignedLogRoot %+v", c.logPrefix, req)
 	}
-
-	request := trillian.GetLatestSignedLogRootRequest{LogId: c.logID}
-	ctx, cancelFunc := context.WithDeadline(context.Background(), getRPCDeadlineTime(c))
-	defer cancelFunc()
-	response, err := c.rpcClient.GetLatestSignedLogRoot(ctx, &request)
-
-	if err != nil || !rpcStatusOK(response.GetStatus()) {
-		return http.StatusInternalServerError, errors.New("backend rpc failed")
+	rsp, err := c.rpcClient.GetLatestSignedLogRoot(ctx, &req)
+	if glog.V(2) {
+		glog.Infof("%s: GetSTH <= grpc.GetLatestSignedLogRoot status=%v", c.logPrefix, rsp.GetStatus())
+	}
+	if err != nil {
+		return http.StatusInternalServerError, fmt.Errorf("backend GetLatestSignedLogRoot request failed: %v", err)
+	}
+	if !rpcStatusOK(rsp.GetStatus()) {
+		return http.StatusInternalServerError, fmt.Errorf("backend GetLatestSignedLogRoot request failed, status=%v", rsp.GetStatus())
 	}
 
-	if treeSize := response.GetSignedLogRoot().TreeSize; treeSize < 0 {
+	// Check over the response.
+	slr := rsp.GetSignedLogRoot()
+	if slr == nil {
+		return http.StatusInternalServerError, fmt.Errorf("no log root returned")
+	}
+	if glog.V(3) {
+		glog.Infof("%s: GetSTH <= slr=%+v", c.logPrefix, slr)
+	}
+	if treeSize := slr.TreeSize; treeSize < 0 {
 		return http.StatusInternalServerError, fmt.Errorf("bad tree size from backend: %d", treeSize)
 	}
 
-	if hashSize := len(response.GetSignedLogRoot().RootHash); hashSize != sha256.Size {
+	if hashSize := len(slr.RootHash); hashSize != sha256.Size {
 		return http.StatusInternalServerError, fmt.Errorf("bad hash size from backend expecting: %d got %d", sha256.Size, hashSize)
 	}
 
-	// Jump through Go hoops because we're mixing arrays and slices, we checked the size above
-	// so it should exactly fit what we copy into it
-	var hashArray [sha256.Size]byte
-	copy(hashArray[:], response.GetSignedLogRoot().RootHash)
-
-	// Build the CT STH object ready for signing
-	sth := ct.SignedTreeHead{TreeSize: uint64(response.GetSignedLogRoot().TreeSize),
-		Timestamp:      uint64(response.GetSignedLogRoot().TimestampNanos / 1000 / 1000),
-		SHA256RootHash: hashArray}
-
-	// Serialize and sign the STH and make sure this succeeds
+	// Build the CT STH object, including a signature over its contents.
+	sth := ct.SignedTreeHead{
+		Version:   ct.V1,
+		TreeSize:  uint64(slr.TreeSize),
+		Timestamp: uint64(slr.TimestampNanos / 1000 / 1000),
+	}
+	copy(sth.SHA256RootHash[:], slr.RootHash) // Checked size above.
```

Comment:
Does this work?
`sth.SHA256RootHash` is `nil`, and `copy` does min(len(dst), len(src)) bytes...